src/AppBundle/Entity/Report/Traits/StatusTrait.php

```diff
@@ -46,28 +60,43 @@ public function setStatusCached(array $status)
         $this->statusCached = json_encode($status);
     }
 
+    /**
+     * @return string
+     */
+    public function getReportStatusCached()
+    {
+        return $this->reportStatusCached;
+    }
+
 
     /**
-     * Update the status cache of the given sections
+     * Update the status cache of the given sections,
+     * and also the report.reportStatusCached using the cache
+     *
      * using the `ReportService::getSectionStateNotCached`
      *
      * @param array $sectionIds
      */
     public function updateSectionsStatusCache(array $sectionIds)
     {
         $currentStatus = $this->getStatusCached();
+        $statusObject = $this->getStatus();
 
-        //TODO optimise and add those only when a financial section is affected
         $sectionIds[] = Report::SECTION_MONEY_TRANSFERS;
         $sectionIds[] = Report::SECTION_BALANCE;
 
         foreach($sectionIds as $sectionId) {
             if ($this->hasSection($sectionId)) {
-                $currentStatus[$sectionId] = $this->getStatus()->getSectionStateNotCached($sectionId);
+                $currentStatus[$sectionId] = $statusObject->getSectionStateNotCached($sectionId);
             }
         }
-
         $this->setStatusCached($currentStatus);
+
+        // update report status, using the cached version of the section statuses
+        // Note: the isDue is skipped
+        $this->reportStatusCached = $statusObject
+            ->setUseStatusCache(true)
+            ->getStatusIgnoringDueDate(true);
```

Comment:
Not sure why the due date is affecting the report status?

Reply:
that's the tricky part. the ReportService.getStatus() is taking into account the end date compared with the today's date. Since the today's data is different every day, then the cached report status will get invalid. What I do to solve is storing without considering the dueDate, and re-add that in the query (indexed way) and also when re hydrating back